build.rs

```diff
@@ -1,66 +1,64 @@
+use std::env;
 use std::path::Path;
 use std::path::PathBuf;
 use std::process::Command;
-use std::{env, fs, io};
 
 #[cfg(feature = "bundled")]
-fn init_submodule(path: &Path) {
+fn fetch_hwloc(path: &Path, version: &str) {
     if !path.join("Makefile.am").exists() {
         let r = Command::new("git")
-            .args(&["clone", "https://github.com/open-mpi/hwloc"])
+            .args(&[
+                "clone",
+                "https://github.com/open-mpi/hwloc",
+                "--depth",
+                "1",
+                "--branch",
+                version,
+            ])
             .current_dir(path.clone())
             .output();
         match r {
-            Ok(m) => {}
-            Err(e) => println!("Error {}", e),
+            Ok(m) => {
+                if !m.status.success() {
+                    eprintln!("git clone for hwloc returned {}", m.status);
+                    eprintln!("{:?}", m);
+                }
+            }
+            Err(e) => eprintln!("Error cloning hwloc {}", e),
```

Comment:
I just realized that as currently written, the code prints an error message... and then goes on to try to build a nonexistent hwloc library ^^'.

Please panic in that case instead. Here's a possible way to do it:

```rust
let m = r.expect("Failed to run git clone");
assert!(
    m.status.success(),
    "git clone for hwloc returned failure code {}\n{:?}",
    m.status(), m
);
```